### PR TITLE
Update use cases and non-functional requirements

### PR DESCRIPTION
**Pull request by ziiqii**
Updated the following for v1.2:
* Use cases
* Non-functional requirements

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -300,45 +300,188 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Use case: UC01 - View help**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to see the help menu.
+1. CulinaryContacts opens up the help window, displaying the command summary.
```
Comment (Tanzhiheng26) [inline docs/DeveloperGuide.md:311]: Should be 2. ?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -300,45 +300,188 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-    Use case ends.
+   Use case ends.
+
+**Use case: UC02 - Add a person**
+
+**MSS**
+
+1. User requests to add a new person.
+1. CulinaryContacts adds the new person to the list.
```
Comment (Tanzhiheng26) [inline docs/DeveloperGuide.md:320]: Should be 2. ?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -300,45 +300,188 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 1b1. CulinaryContacts shows an error message.
+
+      Use case resumes from step 1.
+
+**Use case: UC03 - List all persons**
+
+**MSS**
+
+1. User requests to show all contacts in the list.
+1. CulinaryContacts shows all persons in the list.
```
Comment (Tanzhiheng26) [inline docs/DeveloperGuide.md:343]: Should be 2. ?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -300,45 +300,188 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   Use case ends.
+
+**Use case: UC04 - Edit a person**
+
+**MSS**
+
+1. User requests to <u>list all persons (UC03)</u>.
+1. User requests to edit the fields of a specific person in the list.
+1. CulinaryContacts edits the fields of the person.
```
Comment (Tanzhiheng26) [inline docs/DeveloperGuide.md:353]: Update numbering. Numbers below need to be updated as well.

**Review by zhuuyicheng — APPROVED** [review]
LGTM, approved

**Review by Tanzhiheng26 — CHANGES_REQUESTED** [review]
LGTM, I thought there was an issue with the numbering of lists based on the code but the numbering is correct on the webpage.

**Review by Tanzhiheng26 — APPROVED** [review]

**Pull request merged**